packages/server/src/core/internals/getParseFn.ts

```diff
@@ -30,6 +30,14 @@ export function getParseFn<TType>(procedureParser: Parser): ParseFn<TType> {
     return parser.create.bind(parser);
   }
 
+  if (typeof parser.assert === 'function') {
+    // ParserScaleEsque
+    return (value) => {
+      parser.assert(value);
+      return value as TType;
```

Comment:
Seems like this misses the transform (that you said it supports) since it just returns the original value